commit d7588a2e7fd6895b97d0a347bb69ca0887aa8fbb
Author: Long Li <longli@microsoft.com>
Date:   Wed Dec 14 18:46:01 2016 -0800

    scsi: storvsc: use tagged SRB requests if supported by the device
    
    commit 3cd6d3d9b1abab8dcdf0800224ce26daac24eea2 upstream.
    
    Properly set SRB flags when hosting device supports tagged queuing.
    This patch improves the performance on Fiber Channel disks.
    
    Signed-off-by: Long Li <longli@microsoft.com>
    Reviewed-by: K. Y. Srinivasan <kys@microsoft.com>
    Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linux-4.9.47/drivers/scsi/storvsc_drv.c
===================================================================
--- linux-4.9.47.orig/drivers/scsi/storvsc_drv.c
+++ linux-4.9.47/drivers/scsi/storvsc_drv.c
@@ -103,6 +103,8 @@ struct hv_fc_wwn_packet {
 };
 
 
+#define SP_UNTAGGED			((unsigned char) ~0)
+#define SRB_SIMPLE_TAG_REQUEST		0x20
 
 /*
  * SRB Flag Bits
@@ -1478,6 +1480,13 @@ static int storvsc_queuecommand(struct S
 
 	if (scmnd->device->tagged_supported) {
 		vm_srb->win8_extension.srb_flags |=
+		(SRB_FLAGS_QUEUE_ACTION_ENABLE | SRB_FLAGS_NO_QUEUE_FREEZE);
+		vm_srb->win8_extension.queue_tag = SP_UNTAGGED;
+		vm_srb->win8_extension.queue_action = SRB_SIMPLE_TAG_REQUEST;
+	}
+
+	if (scmnd->device->tagged_supported) {
+		vm_srb->win8_extension.srb_flags |=
 		(SRB_FLAGS_QUEUE_ACTION_ENABLE | SRB_FLAGS_NO_QUEUE_FREEZE);
 		vm_srb->win8_extension.queue_tag = SP_UNTAGGED;
 		vm_srb->win8_extension.queue_action = SRB_SIMPLE_TAG_REQUEST;
Index: linux-4.9.47/dummy/rpi_1266_d7588a2e7fd6895b97d0a347bb69ca0887aa8fbb.txt
===================================================================
--- /dev/null
+++ linux-4.9.47/dummy/rpi_1266_d7588a2e7fd6895b97d0a347bb69ca0887aa8fbb.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
